src/control/agent/main.go

```diff
@@ -52,13 +62,47 @@ func agentMain() error {
 
 	log.Debugf("Starting daos_agent:")
 
-	flag.Parse()
+	p := flags.NewParser(opts, flags.Default)
+
+	_, err := p.Parse()
+	if err != nil {
+		return err
+	}
+
+	// Load the configuration file using the supplied path or the default path if none provided.
+	config, err := client.ProcessConfigFile(opts.ConfigPath)
+	if err != nil {
+		log.Errorf("An unrecoverable error occurred while processing the configuration file: %s", err)
+		return err
+	}
 
-	f, err := common.AppendFile(*logFile)
+	// Apply environment variable overrides if found
+	res := config.ProcessEnvOverrides()
+	log.Debugf("Found %d environment variable overrides", res)
+
+	// Override configuration with any commandline values given
+	err = config.ApplyAgentCmdLineOverrides(opts.RuntimeDir, opts.LogFile)
+	if err != nil {
+		log.Errorf("Failed to apply command line overrides %s", err)
+		return err
+	}
+
+	if config.RuntimeDir == "" {
+		log.Errorf("The path to the agent communications socket is undefined.  "+
+			"Use the config file: %s or command line option 'runtime_dir' to specify a value.", config.Path)
```

Comment:
I think the message is a little unclear - in practice there's a default, so wouldn't they have to have manually set the path to an empty value?

Reply:
You are right that someone would have to set the path to an empty value.  And, because the override is made only if the supplied value is non-empty, then this case cannot actually happen.  Are you OK with me deleting this check and this error message completely?

Reply:
Yeah, makes sense to remove it then.

Reply:
Done.